src/Core/GuiDisplayGen.m

```diff
@@ -419,6 +419,22 @@ - (void) setGLColorFromSetting:(NSString *)setting defaultValue:(OOColor *)def a
 }
 
 
+- (void) setGuiColorSettingFromKey:(NSString *)key color:(OOColor *)col
+{
+	NSMutableDictionary *guiCopy = [guiUserSettings mutableCopy];
+	if (col == nil)
+	{
+		[guiCopy removeObjectForKey:key];
+	} 
+	else
+	{
+		[guiCopy setObject:col forKey:key];
+	}
+	[guiUserSettings release];
+	guiUserSettings = [guiCopy copy];
+}
+
```

Comment:
I would check this part again. I think there is a memory leak the way it is  written above.